logger/src/main/java/com/orhanobut/logger/FileLogger.java

```diff
@@ -0,0 +1,359 @@
+
+        if (MARKERS.contains(message)) {
+            if (BEGIN.equals(message)) {
+
+                stringBuilder.setLength(0);
+                date.setTime(System.currentTimeMillis());
+
+                // machine-readable date/time
+                stringBuilder.append(date.getTime());
+
+                // human-readable date/time
+                stringBuilder.append(separator);
+                stringBuilder.append(dateFormat.format(date));
+
+                // level
+                stringBuilder.append(separator);
+                stringBuilder.append(LEVELS.get(level));
+
+                // tag
+                stringBuilder.append(separator);
+                stringBuilder.append(tag);
+
+
+            } else if (END.equals(message)) {
+
+                stringBuilder.append(NEW_LINE);
+                handler.sendMessage(handler.obtainMessage(0, stringBuilder.toString()));
+
+            }
+            return;
+        }
+
+        if (message.contains(NEW_LINE)) {
+            message = message.replace(NEW_LINE, NEW_LINE_REPLACEMENT);
+        }
+
+        message = message.replace(HORZ_LINE, "").trim();
+
+        // messages
+        stringBuilder.append(separator);
+        stringBuilder.append(message);
+
+    }
+    //endregion
+
+    //region helper to generate file names
+    private static File getLogFile(String folderName, String fileName, int maxLines) {
+
+        File folder = new File(folderName);
+        if (!folder.exists()) {
+            folder.mkdirs();
+        }
+
+        int newFileCount = 0;
+        File newFile;
+        File existingFile = null;
+
+        newFile = new File(folder, String.format("%s_%s.csv", fileName, newFileCount));
+        while (newFile.exists()) {
+            existingFile = newFile;
+            newFileCount++;
+            newFile = new File(folder, String.format("%s_%s.csv", fileName, newFileCount));
+        }
+
+        if (existingFile != null) {
+            try {
+                int lineCount = 0;
+                BufferedReader br = new BufferedReader(new FileReader(existingFile));
+                while (br.readLine() != null) {
+                    lineCount++;
+                }
+
+                if (lineCount >= maxLines) {
+                    return newFile;
+                } else {
+                    return existingFile;
+                }
+            } catch (IOException e) {
+                // don't care
+                e.printStackTrace();
+            }
+        }
+
+        return newFile;
+    }
+    //endregion
+
+    //region handler to write to disk on background thread
+    private class WriteHandler extends Handler {
+        private WriteHandler(Looper looper) {
+            super(looper);
+        }
+
+        @Override
+        public void handleMessage(Message msg) {
+            String text = (String) msg.obj;
+
+            FileWriter fileWriter = null;
+            File logFile = getLogFile(folder, "logs", maxLines);
```

Comment:
pass `folder` and `maxLines` to WriteHandler constructor and make WriteHandler static